homeassistant/components/kodi/media_player.py

```diff
@@ -339,13 +333,16 @@ def async_on_volume_changed(self, sender, data):
 
     async def async_on_quit(self, sender, data):
         """Reset the player state on quit action."""
-        await self._clear_connection()
+        # The websocket connection must be closed. Otherwise the jsonrpc
+        # library throws an uncatchable TransportError exception
+        await self._clear_connection(True)
```

Comment:
This probably belongs in the connection manager rather than the media player

Reply:
✅ As you see below, the "trigger" is a direct reaction to some callback. But since all are only affecting the connection itself I can move those over to the connection manager. My only concern is, that this callbacks could be simply overwritten by future changes in the media_player (sleep maybe).
```Python
    async def async_on_quit(self, sender, data):
        """Reset the player state on quit action."""
        # The websocket connection must be closed. Otherwise the jsonrpc
        # library throws an uncatchable TransportError exception
        await self._clear_connection(True)

[...]
        server.System.OnQuit = self.async_on_quit
        server.System.OnRestart = self.async_on_quit
        server.System.OnSleep = self.async_on_quit
```

Reply:
All callbacks could be overwritten, but I still think the connection manager is the right place for these. The media player should handle media player-y stuff :)